sdk/eventhub/azure-eventhubs/azure/eventhub/producer.py

```diff
@@ -109,6 +112,10 @@ def _open(self):
             self._connect()
             self.running = True
 
+            self._max_message_size_on_link = self._handler.message_handler._link.peer_max_message_size if\
+                self._handler.message_handler._link.peer_max_message_size\
```

Comment:
Under what circumstances there is no peer_max_message_size? When the link is not open?

Reply:
I'm not sure about it. It seems that the peer_max_message_size should always be returned back from service side as it's part of the link negotiation in amqp protocol.
This piece of code at least gives a local default value.